Core/src/org/sleuthkit/autopsy/directorytree/actionhelpers/ExtractActionHelper.java

```diff
@@ -0,0 +1,357 @@
+         *
+         * @param extractionTasks List of file extraction tasks.
+         */
+        FileExtracter(List<FileExtractionTask> extractionTasks) {
+            this.extractionTasks = extractionTasks;
+        }
+
+        @Override
+        protected Object doInBackground() throws Exception {
+            if (extractionTasks.isEmpty()) {
+                return null;
+            }
+
+            // Setup progress bar.
+            final String displayName = NbBundle.getMessage(this.getClass(), "ExtractActionHelper.progress.extracting");
+            progress = ProgressHandle.createHandle(displayName, new Cancellable() {
+                @Override
+                public boolean cancel() {
+                    if (progress != null) {
+                        progress.setDisplayName(
+                                NbBundle.getMessage(this.getClass(), "ExtractActionHelper.progress.cancellingExtraction", displayName));
+                    }
+                    return ExtractActionHelper.FileExtracter.this.cancel(true);
+                }
+            });
+            progress.start();
+            progress.switchToIndeterminate();
+
+            /*
+             * @@@ Add back in -> Causes exceptions int workUnits = 0; for
+             * (FileExtractionTask task : extractionTasks) { workUnits +=
+             * calculateProgressBarWorkUnits(task.source); }
+             * progress.switchToDeterminate(workUnits);
+             */
+            // Do the extraction tasks.
+            for (FileExtractionTask task : this.extractionTasks) {
+                // @@@ Note, we are no longer passing in progress
+                ContentUtils.ExtractFscContentVisitor.extract(task.source, task.destination, null, this);
+            }
+
+            return null;
+        }
+
+        @Override
+        protected void done() {
+            boolean msgDisplayed = false;
+            try {
+                super.get();
+            } catch (InterruptedException | ExecutionException ex) {
+                logger.log(Level.SEVERE, "Fatal error during file extraction", ex); //NON-NLS
+                MessageNotifyUtil.Message.info(
+                        NbBundle.getMessage(this.getClass(), "ExtractActionHelper.done.notifyMsg.extractErr", ex.getMessage()));
+                msgDisplayed = true;
+            } finally {
+                progress.finish();
+                if (!this.isCancelled() && !msgDisplayed) {
+                    MessageNotifyUtil.Message.info(
+                            NbBundle.getMessage(this.getClass(), "ExtractActionHelper.done.notifyMsg.fileExtr.text"));
+                }
+            }
+        }
+
+        /**
+         * Calculate the number of work units for the progress bar.
+         *
+         * @param file File whose children will be reviewed to get the number of
+         *             work units.
+         *
+         * @return The number of work units.
+         */
+        /*
```

Comment:
Do you need this commented out function?

Reply:
I left it as it was due to the @@@ style todo at line 306 which says to add this back in.